test/credentials.spec.coffee

```diff
@@ -832,3 +840,26 @@ describe 'AWS.CognitoIdentityCredentials', ->
         creds.getId (err, id) ->
           expect(id).to.equal('IDENTITY-ID2')
           expect(creds.getStorage('id')).to.equal('IDENTITY-ID2')
+
+      it 'returns cached id in getId call with LoginId', ->
+        setupClients LoginId: 'LOGINIDA'
+        helpers.mockResponses [
+          {data: {IdentityId: 'IDENTITY-ID1'}, error: null},
+          {data: {TOKEN: 'TOKEN', IdentityId: 'IDENTITY-ID2'}, error: null}
+        ]
+        helpers.spyOn(creds.webIdentityCredentials, 'refresh').andCallFake (cb) ->
+          creds.webIdentityCredentials.data =
+            Credentials:
+              AccessKeyId: 'KEY'
+              SecretAccessKey: 'SECRET'
+              SessionToken: 'TOKEN'
+          cb null
+        creds.refresh(->)
+        creds.getId (err, id) ->
+          expect(id).to.equal('IDENTITY-ID2')
+          expect(creds.getStorage('id')).to.equal('IDENTITY-ID2')
+        setupCreds({LoginId: 'LOGINIDB'})
+        expect(creds.getStorage('id')).not.to.equal('IDENTITY-ID2')
+        setupCreds({LoginId: 'LOGINIDA'})
```

Comment:
These last 2 steps are actually a clean-up step.  When testing in a browser environment, these credentials would be cached, so the tests would pass the first time it was run in a browser, but not subsequent times. I could explicitly delete the credentials form localstorage instead, so it's more clear why I'm doing this.